task-maker-async/src/store.rs

```diff
@@ -0,0 +1,98 @@
+#![allow(dead_code)]
+
+use serde::{Deserialize, Serialize};
+
+type HashData = Vec<u8>;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InputFileHash(HashData);
+
+/// Two-level hash; the outer hash has information about all properties of a computation that are
+/// expected to change the result (such as outer hashes of inputs, or the command line). The inner
+/// hash takes care of properties of the computation that should not change the outputs, such as
+/// time and memory limits; it also includes the first hash.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ComputationHash(HashData, HashData);
```

Comment:
I get the point of having two hashes for this. I'm still not so sure on how `similar_computations` should be used though. It returns the set of `ComputationHash` sharing the first of the two hashes. But how can one get the information used for computing those hashes? I guess they are written somewhere as metadata file associated to a `ComputationHash`, but where?

Reply:
I added a second type of special files, other than MainFile, for `FileSet`s - a `Metadata` file type that will store this kind of information.